### DISCUSSION TITLE
[New Feature] Money send to ally system

### DISCUSSION THREAD

**Discussion by DuisPlay** [Vinifera-Developers/Vinifera#1317] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

In some C&C Generals mods you can send money to your ally. To do this, players can build delivery trucks in their supply center and send them to their allies, where the truck docks at the allied supply center (the ally then receives $ 500 and the truck despawn).

### Proposed Code (optional)

To define the vehicle and the reception building, the following could be used in the rules.ini:

[VehicleType]
IsSupplyTruck=yes/no or true/false
SupplyDock=BuildingType
FinancialSupportHigh=amount

**Comment by JoyfulShush** [comment]
While this is a known method to do this, I think it's pretty limited - it forces modders to make a buildable delivery truck unit and a supply center building (or designate buildings, such as ConYard to be a supply center). It also means that the system is very specific and might not be relevant for some mods or maps.

Instead, how about a small button that opens a window (think about the debug window, for instance) where you can click on a player that you are allied to in order to give it 500 or 1000 credits per click, assuming you have enough to afford it.

**Reply by Templarfreak** [reply]
this is in general how other RTS games other than C&C handle sending resources to an ally. i do *like* the idea of supply trucks because it means they are harassable and you have to defend them as they go to your ally, but it's also a lot more effort to do something in a game where you're kind of already expected to have the possibility of virtually infinite income anyway (well, not Generals, but other C&C games yeah) and thus resource trading as a concept is already in this position where a lot of people arent very interested in it anyway

**Reply by Longshanks1** [reply]
This could work (without the need for the supply center) with PR #1157 [Customizable Crates]. It could look something like this

[SUPPLYTRUCK]
CarriesCrate=yes
CrateToSpawn=Money
AmountToSpawn=3000

Build the truck, send it to your ally's base, then deploy it to self destruct the truck and drop the crate

**Reply by JoyfulShush** [reply]
@Templarfreak I think the main use case of this mechanism is one player being left with no money and no refineries, and so has no way to stay in the game as they cannot rebuild. In such cases your team is already behind, and having to build units just to give your ally moeny in small increments instead of making more army units means you'll probably be losing the game due to it.
